snapshots/overlay/overlay.go

```diff
@@ -95,11 +96,18 @@ func NewSnapshotter(root string, opts ...Opt) (snapshots.Snapshotter, error) {
 		indexOff = true
 	}
 
+	// figure out whether "userxattr" option is recognized by the kernel && needed
+	userxattr, err := NeedsUserXAttr(root)
+	if err != nil {
+		return nil, err
```

Comment:
Could we just ignore the error here to ensure that this is very safe?

Reply:
There is already quite a few errors ignored in the function and should only trigger errors if already in the user namespace.

@AkihiroSuda if we were to always ignore errors here, will it always error out on running containers in 5.11 and later? Plugin load errors are rare, but would that be less confusing than just having the mount errors later on.

Reply:
> if we were to always ignore errors here, will it always error out on running containers in 5.11 and later? 

- rootful is unaffected
- Yes for most rootless containers, but not "always"

Reply:
I think the cleanest solution here is to just have the `NeedsUserXattr` return errors if it wasn't able to make that determination then just warn about that error here. Considering it is just a flag which gets put on the mount, there shouldn't be a need to fail loading the whole overlay plugin. It is reasonable to log that at warn or error level though, since it shouldn't occur under normal circumstances.

Reply:
changed `return err` to `logrus.Warn`